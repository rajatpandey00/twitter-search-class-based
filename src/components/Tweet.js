import React from 'react'
import Card from 'react-bootstrap/Card'

export default ({ tweet, tkey }) => {
    return(
    <Card className="tweet" key={tkey}>
        <img
            width={32}
            height={32}
            className="mr-3"
            src={tweet?.user?.profile_image_url || tweet?.user?.profile_image_url_https}
        />
        <Card.Body key={tkey}>
            <Card.Text key={tkey}>
                {tweet?.text}
            </Card.Text>
        </Card.Body>
        <Card.Footer>
            <span key={tkey} style={{float: "right"}}>Retweeted Count - {tweet?.retweet_count}</span>
        </Card.Footer>
    </Card>)
}