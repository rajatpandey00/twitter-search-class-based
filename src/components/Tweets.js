import React from 'react'
import Tweet from './Tweet';

import '../styles/search.scss';

// @flow
export default ({ tweets }) => {
    return (
        <div>
            {tweets.length > 0 && tweets.map((tweet, index) => <Tweet tweet={tweet} tkey={index} key={index} />)}
        </div>
    )
}