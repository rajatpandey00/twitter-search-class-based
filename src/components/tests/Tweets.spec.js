import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Tweets from '../Tweets';
import Tweet from '../Tweet'

let mockTweets;
describe('Tweets test', () => {
  mockTweets = [{ text: 'test tweet', retweet_count: 9 }]
  test('snapshot renders', () => {
    const component = renderer.create(<Tweets tweets={mockTweets} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Tweets test', () => {
  mockTweets = [{ text: 'test tweet', retweet_count: 9 }, { text: 'test tweet2', retweet_count: 19 }]
  const wrapper = shallow(<Tweets tweets={mockTweets} />);
  it.only('should render separate div', () => {
    expect(wrapper.find(Tweet).length).toEqual(mockTweets.length);
  });
});