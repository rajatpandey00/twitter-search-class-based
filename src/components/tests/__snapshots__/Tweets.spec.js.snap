// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweets test snapshot renders 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "cadetblue",
          "width": "50rem",
        }
      }
    >
      <img
        className="mr-3"
        height={32}
        width={32}
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          test tweet
        </p>
      </div>
      <div
        className="card-footer"
      >
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          Retweeted Count - 
          9
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "cadetblue",
          "width": "50rem",
        }
      }
    >
      <img
        className="mr-3"
        height={32}
        width={32}
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          test tweet2
        </p>
      </div>
      <div
        className="card-footer"
      >
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          Retweeted Count - 
          19
        </span>
      </div>
    </div>
  </div>
</div>
`;
